components/ntp_sponsored_images/browser/ntp_sponsored_images_component_manager.h

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_NTP_SPONSORED_IMAGES_BROWSER_NTP_SPONSORED_IMAGES_COMPONENT_MANAGER_H_
+#define BRAVE_COMPONENTS_NTP_SPONSORED_IMAGES_BROWSER_NTP_SPONSORED_IMAGES_COMPONENT_MANAGER_H_
+
+#include <memory>
+#include <string>
+
+#include "base/files/file_path.h"
+#include "base/memory/weak_ptr.h"
+#include "base/observer_list.h"
+#include "base/optional.h"
+#include "brave/components/brave_component_updater/browser/brave_component.h"
+#include "brave/components/ntp_sponsored_images/browser/ntp_sponsored_image_source.h"
+#include "components/component_updater/component_updater_service.h"
+
+struct NTPSponsoredImagesData;
+struct NTPSponsoredImagesInternalData;
+
+namespace content {
+class BrowserContext;
+}  // namespace content
+
+// NOTE: Client should call AddDataSource() before requesting image sources.
+// When component is updated, Client also should call AddDataSource() for add
+// proper |URLDataSource|s.
+class NTPSponsoredImagesComponentManager
+    : public brave_component_updater::BraveComponent,
+      public component_updater::ServiceObserver  {
+ public:
+  class Observer {
+   public:
+    // Called whenever ntp sponsored images component is updated.
+    virtual void OnUpdated(const NTPSponsoredImagesData& data) = 0;
+   protected:
+    virtual ~Observer() {}
+  };
+
+  NTPSponsoredImagesComponentManager(
+      BraveComponent::Delegate* delegate,
+      component_updater::ComponentUpdateService* cus);
+  ~NTPSponsoredImagesComponentManager() override;
+
+  NTPSponsoredImagesComponentManager(
```

Comment:
use `DISALLOW_COPY_AND_ASSIGN`

Reply:
OK to leave as-is, IMO. See note at https://github.com/brave/brave-core/pull/4075/files/2f7f51f8a0ffee135d823311f7bb464dd74e6acc#r370352211